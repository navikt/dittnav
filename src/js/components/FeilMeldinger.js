import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage as F } from 'react-intl';

class FeilMeldinger extends Component {
  render() {
    return (
      <React.Fragment>
        {this.props.errors.length > 0 &&
        <section className="service-notification">
          <h1 className="typo-element">
            {this.props.errors.map(e => (<span key={e}><F id={e} /></span>))}
          </h1>
        </section>}
      </React.Fragment>
    );
  }
}

FeilMeldinger.propTypes = {
  errors: PropTypes.arrayOf(PropTypes.string),
};

FeilMeldinger.defaultProps = {
  errors: [],
};

export default FeilMeldinger;
