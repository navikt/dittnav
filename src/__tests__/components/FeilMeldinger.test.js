import * as React from 'react';
import FeilMeldinger from 'js/components/FeilMeldinger';
import wrapIntl from 'js/IntlTestHelper';
const ReactTestRenderer = require('react-test-renderer');

test('render empty FeilMeldinger', () => {
  const component = ReactTestRenderer.create(wrapIntl(<FeilMeldinger />));
  expect(component.toJSON()).toMatchSnapshot();
});

test('render one error FeilMeldinger', () => {
  const component = ReactTestRenderer.create(wrapIntl(<FeilMeldinger errors={['error.general.connection.problem']} />));
  expect(component.toJSON()).toMatchSnapshot();
});
