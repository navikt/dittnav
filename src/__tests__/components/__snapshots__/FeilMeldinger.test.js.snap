// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render empty FeilMeldinger 1`] = `null`;

exports[`render one error FeilMeldinger 1`] = `
<section
  className="service-notification"
>
  <h1
    className="typo-element"
  >
    <span>
      <span>
        Error fetchin data
      </span>
    </span>
  </h1>
</section>
`;
